### PR TITLE
Add AWS Integrations section

### PR DESCRIPTION
**Pull request by bilbof**
This lists the cluster-level integrations with AWS services we have installed.

https://trello.com/c/1k2eVCZS/670-good-enough-user-docs

### CODE REVIEW

File: source/documentation/index.md [hunk 1]
```diff
@@ -282,3 +282,25 @@ We recommend using [tab completion](https://kubernetes.io/docs/reference/kubectl
...
+The Kubernetes cluster interacts with AWS services to provide features
+such as DNS, ALB ingress, and Secret provisioning.
+Here we list the operators installed in the cluster that integrate with AWS.
+
+* Secrets: The cluster uses the [external-secrets] operator to access secrets
+in AWS Secrets Manager and automatically injects the values as Kubernetes Secrets.
+* DNS: [external-dns] is a Kubernetes addon that configures public DNS servers
+(in our case AWS Route53) with information about exposed Kubernetes services
+to make them discoverable.
+* ALB Ingress: [aws-load-balancer-controller] provisions AWS ALBs for Kubernetes
```
Comment (jonathanglassman) [inline source/documentation/index.md:297]: Do users know what ALBs and NLBs are? Or do we need to spell out these acronyms?

**Review by jonathanglassman — COMMENTED** [review]
Looks good, made one comment.

File: source/documentation/index.md (hunk 1 above)
Comment (bilbof) [reply source/documentation/index.md:297]: They should do yes, I don't think we need to spell it out for this audience.

Comment (jonathanglassman) [reply source/documentation/index.md:297]: 👍

**Review by jonathanglassman — APPROVED** [review]

**Pull request merged**